### ISSUE TITLE
Export C Header doesn't respect ENUM size

### ISSUE THREAD

**Issue by h0x91b** [NationalSecurityAgency/ghidra#1840]
**Describe the bug**

If you define 1 byte sized ENUM in ghidra then export it using "Export C Header" context menu function, it exported as regular ENUM 4 bytes length. Size of ENUM isn't respected by exporter.

e.g.

```c
typedef enum SPRITE_BIT1 {
    SPRITE_BIT1_1=1
} SPRITE_BIT1;
```

Due to this bug, all offsets in exported structs are shifted, currently, I am fixing exported H file manually...

**To Reproduce**

Define new ENUM, select any non-4 bytes size, then export it,

**Expected behavior**

1 Bytes length ENUMs should be exported as

```c
typedef enum SPRITE_BIT1 : unsigned char {
    SPRITE_BIT1_1=1
} SPRITE_BIT1;
```

2 Bytes length ENUMs should be exported as
```c
typedef enum SPRITE_BIT1 : unsigned short {
    SPRITE_BIT1_1=1
} SPRITE_BIT1;
```
**Environment (please complete the following information):**
 - OS:  Windows 10
 - Java Version: 12.0.1
 - Ghidra Version: 9.1.2

**Comment by mumbel** [comment]
That syntax is not in C (introduced in C++11?)

**Comment by h0x91b** [comment]
Do not know, maybe... I am using it in std:c++17

**Comment by emteere** [comment]
I suppose export options for strict C or C++ export flavor could be added that would address this issue.
The default could be the most common compiler.

**Comment by rafa-br34** [comment]
> I suppose export options for strict C or C++ export flavor could be added that would address this issue. The default could be the most common compiler.

Yeah, the export to C feature is absurdly limited, I just spent a good part of an hour trying to figure out why something wasn't working just to find out that it was because the supposedly u8 enum was in fact using up +24 bits of the structure...
Somewhat related to #993
